app/Jobs/Imports/ImportRockTheVoteRecord.php

```
@@ -0,0 +1,679 @@
+            'northstar_id' => $user->id,
+            'source_details' => $this->record->postData['source_details'],
+            'type' => (string) PostType::VOTER_REG(),
+        ];
+
+        $signup = $signups->get($user->id, $action->campaign->id);
+
+        if (!$signup) {
+            $signup = $signups->create(
+                $payload,
+                $user->id,
+                $action->campaign->id,
+            );
+        }
+
+        $post = $posts->create($payload, $signup->id);
+
+        info('ImportRockTheVoteRecord - created post', [
+            'user_id' => $user->id,
+            'action_id' => $action->id,
+            'post_id' => $post->id,
+        ]);
+
+        return $post;
+    }
+
+    /**
+     * Updates post with import record data if needed.
+     *
+     * @param \App\Models\Post $post
+     * @return array
+     */
+    private function updatePost(Post $post)
+    {
+        $shouldUpdateStatus = self::shouldUpdateStatus(
+            $post['status'],
+            $this->record->postData['status'],
+        );
+
+        if (!$shouldUpdateStatus) {
+            return $post;
+        }
+
+        $post->fill([
+            'status' => $this->record->postData['status'],
+        ]);
+
+        info('ImportRockTheVoteRecord - updated post', [
+            'post_id' => $post->id,
+            'status' => $post->status,
+        ]);
+
+        return $post;
+    }
+
+    /**
+     * Import the Rock The Vote record as a new user.
+     *
+     * @param \App\Managers\SignupManager $signup
+     * @param \App\Managers\PostManager $post
+     */
+    private function importRecordAsNewUser($signups, $posts)
+    {
+        $user = $this->registrar->register($this->record->userData);
+
+        $action = Action::find($this->record->postData['action_id']);
+
+        $post = $this->createPost($user, $action, $signups, $posts);
+
+        if (!$post) {
+            return [];
+        }
```

Is it fine to return empty array or maybe throw an error?